### PR TITLE
Updated samples for the Adobe PDF Services SDK 3.0.0

### PR DESCRIPTION
**Pull request by vardhi**
This PR aims at the following:
1. Updated all the samples to use the latest Adobe PDF Services SDK version 3.0.0.
2. Added a sample, ExportPDFToDOCXWithOCROption, to showcase the support of OCR locales for Export PDF.
3. Added new samples, ExportPDFToJPEG and ExportPDFToJPEGZip, to illustrate the retrieval of either a list or a zip archive of JPEG images from the specified PDF file.
4. Added a sample, GetPDFProperties, to illustrate the retrieval of various properties of the specified PDF file.

### CODE REVIEW

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java [hunk 1]
```diff
@@ -43,17 +48,23 @@ public static void main(String[] args) {
...
-            // Provide an input FileRef for the operation
-            FileRef source = FileRef.createFromLocalFile("src/main/resources/pdfPropertiesInput.pdf");
-            pdfPropertiesOperation.setInputFile(source);
+            // Execute the operation.
+            List<FileRef> results = exportPDFToImagesOperation.execute(executionContext);
 
-            // Execute the operation
-            FileRef result = pdfPropertiesOperation.executeAndReturnFileRef(executionContext);
+            //to obtain the media type of asset
+            LOGGER.info("media type of the received asset is "+results.get(0).getMediaType());
```
Comment (ankur010592) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java:64]: What is the use case of adding a log line here?

**Review by ankur010592 — CHANGES_REQUESTED** [review]

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java [hunk 2]
```diff
@@ -8,59 +8,62 @@
...
             ExecutionContext executionContext = ExecutionContext.create(credentials);
-            ExportPDFOperation exportPdfOperation = ExportPDFOperation.createNew(ExportPDFTargetFormat.JPEG);
+            ExportPDFOperation exportPdfOperation = ExportPDFOperation.createNew(ExportPDFTargetFormat.DOCX);
 
             // Set operation input from a source file.
-            FileRef sourceFileRef = FileRef.createFromLocalFile("src/main/resources/exportPDFToImageInput.pdf");
+            FileRef sourceFileRef = FileRef.createFromLocalFile("src/main/resources/exportPDFInput.pdf");
             exportPdfOperation.setInput(sourceFileRef);
 
+            //Set export pdf ocr option
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java:54]: Create a new ExportPDFOptions instance from the specified OCR locale and set it into the operation

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java [hunk 3]
```diff
@@ -8,59 +8,62 @@
...
 
             // Initial setup, create credentials instance.
             Credentials credentials = Credentials.serviceAccountCredentialsBuilder()
                     .fromFile("pdfservices-api-credentials.json")
                     .build();
-
             //Create an ExecutionContext using credentials and create a new operation instance.
             ExecutionContext executionContext = ExecutionContext.create(credentials);
-            ExportPDFOperation exportPdfOperation = ExportPDFOperation.createNew(ExportPDFTargetFormat.JPEG);
+            ExportPDFOperation exportPdfOperation = ExportPDFOperation.createNew(ExportPDFTargetFormat.DOCX);
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java:48]: variable name should be exportPDFOperation

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java [hunk 4]
```diff
@@ -8,59 +8,62 @@
...
+            //Set export pdf ocr option
+            ExportPDFOptions exportPDFOptions = new ExportPDFOptions(ExportOCRLocale.EN_US);
+            exportPdfOperation.setOptions(exportPDFOptions);
+
             // Execute the operation.
             FileRef result = exportPdfOperation.execute(executionContext);
 
             // Save the result to the specified location.
-            result.saveAs("output/exportPDFToJPEG.zip");
+            result.saveAs("output/exportPdfOutputWithOCR.docx");
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java:62]: exportPDFWithOCROptionsOutput.docx

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java [hunk 5]
```diff
@@ -8,59 +8,62 @@
...
 
         } catch (ServiceApiException | IOException | SdkException | ServiceUsageException ex) {
             LOGGER.error("Exception encountered while executing operation", ex);
         }
     }
+
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdf/ExportPDFToDOCXWithOCROption.java:68]: Please remove this extra line.

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java [hunk 6]
```diff
@@ -43,17 +48,23 @@ public static void main(String[] args) {
...
+            FileRef sourceFileRef = FileRef.createFromLocalFile("src/main/resources/exportPDFToImagesInput.pdf");
+            exportPDFToImagesOperation.setInput(sourceFileRef);
+
+            // Set the output type to create zip of images
+            exportPDFToImagesOperation.setOutputType(OutputType.ZIP_OF_PAGE_IMAGES);
 
-            // Provide an input FileRef for the operation
-            FileRef source = FileRef.createFromLocalFile("src/main/resources/pdfPropertiesInput.pdf");
-            pdfPropertiesOperation.setInputFile(source);
+            // Execute the operation.
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java:60]: Let us either keep a full stop everywhere in the file comments, or remove it from everywhere in the file comments.

File: src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java [hunk 7]
```diff
@@ -47,9 +59,14 @@ public static void main(String[] args) {
                     .build();
             pdfPropertiesOperation.setOptions(pdfPropertiesOptions);
 
-            // Execute the operation and return JSON Object
-            JSONObject result = pdfPropertiesOperation.execute(executionContext);
-            LOGGER.info("The resultant PDF Properties are: {}", result);
+            // Execute the operation pdf properties operation
+            PDFProperties result = pdfPropertiesOperation.execute(executionContext);
+            // get properties of the pdf
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java:64]: Put an extraline before this comment.

File: src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java [hunk 8]
```diff
@@ -47,9 +59,14 @@ public static void main(String[] args) {
                     .build();
             pdfPropertiesOperation.setOptions(pdfPropertiesOptions);
 
-            // Execute the operation and return JSON Object
-            JSONObject result = pdfPropertiesOperation.execute(executionContext);
-            LOGGER.info("The resultant PDF Properties are: {}", result);
+            // Execute the operation pdf properties operation
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java:62]: Execute the operation

File: src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java (hunk 7 above)
Comment (anubhav-mhjn) [reply src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java:64]: Fetch the requisite properties of the specified PDF

File: src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java [hunk 9]
```diff
@@ -47,9 +59,14 @@ public static void main(String[] args) {
...
+            PDFProperties result = pdfPropertiesOperation.execute(executionContext);
+            // get properties of the pdf
+            LOGGER.info("The Page level properties of the PDF: {}", result.getDocument().getPageCount());
+            LOGGER.info("The Fonts used in the PDF: ");
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/pdfproperties/GetPDFProperties.java:66]: Put an extra line before this log statement

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java [hunk 10]
```diff
@@ -43,17 +48,23 @@ public static void main(String[] args) {
...
             //Create an ExecutionContext using credentials and create a new operation instance.
             ExecutionContext executionContext = ExecutionContext.create(credentials);
-            PDFPropertiesOperation pdfPropertiesOperation = PDFPropertiesOperation.createNew();
+            ExportPDFToImagesOperation exportPDFToImagesOperation = ExportPDFToImagesOperation.createNew(ExportPDFToImagesTargetFormat.JPEG);
+
+            // Set operation input from a source file.
+            FileRef sourceFileRef = FileRef.createFromLocalFile("src/main/resources/exportPDFToImagesInput.pdf");
+            exportPDFToImagesOperation.setInput(sourceFileRef);
+
+            // Set the output type to create zip of images
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java:57]: Set the operation output type as Zip of images

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java [hunk 11]
```diff
@@ -43,17 +48,23 @@ public static void main(String[] args) {
...
 
-            // Execute the operation
-            FileRef result = pdfPropertiesOperation.executeAndReturnFileRef(executionContext);
+            //to obtain the media type of asset
+            LOGGER.info("media type of the received asset is "+results.get(0).getMediaType());
 
-            // Save the result at the specified location
-            result.saveAs("output/pdfPropertiesOutput.json");
+            // Save the result to the specified location.
+            results.get(0).saveAs("output/exportPDFToJPEG.zip");
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java:67]: exportPDFToJPEGOutput.zip

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java [hunk 12]
```diff
@@ -43,17 +48,23 @@ public static void main(String[] args) {
...
 
-            // Provide an input FileRef for the operation
-            FileRef source = FileRef.createFromLocalFile("src/main/resources/pdfPropertiesInput.pdf");
-            pdfPropertiesOperation.setInputFile(source);
+            // Execute the operation.
+            List<FileRef> results = exportPDFToImagesOperation.execute(executionContext);
 
-            // Execute the operation
-            FileRef result = pdfPropertiesOperation.executeAndReturnFileRef(executionContext);
+            //to obtain the media type of asset
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEGZip.java:63]: Is there any explicit reqt to keep this logging statement?

File: src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEG.java [hunk 13]
```diff
@@ -30,10 +30,10 @@
  * <p>
  * Refer to README.md for instructions on how to run the samples.
  */
-public class ExportPDFToJPEGList {
+public class ExportPDFToJPEG {
 
     // Initialize the logger.
-    private static final Logger LOGGER = LoggerFactory.getLogger(ExportPDFToJPEGList.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportPDFToJPEG.class);
```
Comment (anubhav-mhjn) [inline src/main/java/com/adobe/pdfservices/operation/samples/exportpdftoimages/ExportPDFToJPEG.java:36]: We can change the output naming for this class from exportPDFToImagesOutput_ to exportPDFToJPEGOutput_

**Review by anubhav-mhjn — CHANGES_REQUESTED** [review]

**Review by ankur010592 — APPROVED** [review]

**Review by anubhav-mhjn — APPROVED** [review]

**Review by pulkit-maheshwr — APPROVED** [review]

**Pull request merged**